src/components/sidebar/Sidebar.tsx

```diff
@@ -82,24 +86,17 @@ export default function Sidebar() {
   const [hoveredButton, setHoveredButton] = useState<string | null>(null);
 
   const { isAuthenticated } = useAuth();
-  const { voyageTeamMembers } = useUser();
-
-  const isActive = useMemo(() => {
-    if (voyageTeamMembers.length === 0) {
-      return false;
-    }
-    return voyageTeamMembers.some(
-      (member) => member.voyageTeam.voyage.status.name === "Active",
-    );
-  }, [voyageTeamMembers]);
-
-  const isVoyageStarted: boolean = isAuthenticated && isActive;
-
-  const currentVoyageTeam = voyageTeamMembers.find(
-    (voyage) => voyage.voyageTeam.voyage.status.name === "Active",
+  const user = useUser();
+  const voyageTeamAdapter = resolve<VoyageTeamClientAdapter>(
```

Comment:
Definitely. I did have something like this in mind but just implemented like this because I'm assuming the way these classes get instantiated will be changed later anyways. Another reason I didn't do it now is because I wasn't entirely sure what would be a good way to do it until I see how the final structure will be with the entire modules folder that will be moved to the separate repo

Reply:
> Definitely. I did have something like this in mind but just implemented like this because I'm assuming the way these classes get instantiated will be changed later anyways. Another reason I didn't do it now is because I wasn't entirely sure what would be a good way to do it until I see how the final structure will be with the entire modules folder that will be moved to the separate repo

@Dan-Y-Ko In my company's app we have dozens of UseCases (adapters in chingu's case), and the logic that you see in my previous comment is what we actually do

Reply:
> > Definitely. I did have something like this in mind but just implemented like this because I'm assuming the way these classes get instantiated will be changed later anyways. Another reason I didn't do it now is because I wasn't entirely sure what would be a good way to do it until I see how the final structure will be with the entire modules folder that will be moved to the separate repo
> 
> @Dan-Y-Ko In my company's app we have dozens of UseCases (adapters in chingu's case), and the logic that you see in my previous comment is what we actually do

I'm pretty sure this would be part of the refactors you wanted to do with how the adapters are called. I'm doing it this same way in all the other files

From previous pr:
![screenshot](https://i.imgur.com/kuuw8kJ.png)